src/main/java/teammates/common/datatransfer/StudentAttributes.java

```diff
@@ -130,7 +129,7 @@ public boolean isRegistered() {
     }
 
     public String getRegistrationUrl() {
-        return new Url(Config.APP_URL + Const.ActionURIs.STUDENT_COURSE_JOIN_NEW)
+        return new Url(Const.ActionURIs.STUDENT_COURSE_JOIN_NEW)
```

Comment:
1. I couldn't test the redirects before. I need either the URL to be relative or the app URL in `build.properties` to be enforced to be a working URL. The latter will render God mode unavailable (because then `app.url` and `test.url` must always be the same) and is also against what is written (`For dev server, any URL would do.`), so it's quite clear that the former (i.e this fix) is the way to go.
2. Relative URL is _already_ the norm in the system. There are only very few exceptions: this case (fixed), the feedback session copy modal (should be fixed), the three I mentioned above (emails, admin search, admin home page right after account creation; these three use absolute URLs correctly), and one in `AdminEmailImageUploadAction` (haven't investigated whether this can be relative, and certainly not here if I'm going to). Everything else other than those are relative URLs.

In other words, yes I can defer the `PageData.java` fix to another issue, but everything else needs to be in this fix.